extensions/amp-story/1.0/amp-story.js

```diff
@@ -1835,6 +1855,29 @@ export class AmpStory extends AMP.BaseElement {
         containsMediaElementWithAudio || hasStoryAudio);
   }
 
+  /**
+   * Checks for the presence of a sidebar. If a sidebar does exist, then an icon
+   * permitting for the opening/closing of the sidebar is shown.
+   * @private
+   */
+  checkForSidebar_() {
+    this.sidebar_ = this.element.querySelector('amp-sidebar');
+    if (this.sidebar_) {
+      this.storeService_.dispatch(Action.TOGGLE_STORY_HAS_SIDEBAR,
+          !!this.sidebar_);
+      const actions = Services.actionServiceForDoc(this.getAmpDoc());
+      actions.addToWhitelist('AMP-SIDEBAR.open');
+      actions.addToWhitelist('AMP-SIDEBAR.close');
+      actions.addToWhitelist('AMP-SIDEBAR.toggle');
+      const sidebarObserver = new this.win.MutationObserver(() => {
```

Comment:
Browser support might be good for this, but if the feature is unsupported, user would be stuck with a paused story.
Should we do a feature check, and unpause the story right away if `MutationObserver` is unsupported?

These users would have the story playing under the sidebar, but that's better than being stuck.

````
if (this.win.MutationObserver) {
  // Your code.
} else {
  this.storeService_.dispatch(Action.TOGGLE_PAUSED, false);
}
````

Also, I'm wondering whether we should only set the MutationObserver when the sidebar is open. We can use `observe` to start listening, but also `disconnect` to stop listening. Do you think it'd make sense? If that'd make the code too complex I think it's fine to keep it.

Reply:
I guess also to check: are we sure the sidebar itself doesn't trigger an event when it's opened/closed?  Should it?

Reply:
I mostly addressed @gmajoulet's comment in the comment under the combination of toggling of the PAUSED and SIDEBAR state.

in regards to @newmuis through debugging, I interpreted the sidebar object itself not triggering an event, but TOGGLE_SIDEBAR is dispatched both on the click of the system layer button and in the mutation observer. the feature works, but as explained in the same above comment, I haven't yet found a way to incorporate all feedback while avoiding this. 

Any suggestions would be appreciated :)

Reply:
`const sidebarObserver = new this.win.MutationObserver(() => {` is going to fail on a browser where MutationObserver is not defined. So `if(!this.win.MutationObserver) {` would never be reached